x-pack/legacy/plugins/siem/public/components/header_global/index.test.tsx

```diff
@@ -18,8 +20,20 @@ jest.mock('../search_bar', () => ({
   SiemSearchBar: () => null,
 }));
 
+const mockHostsLocation = {
+  pathname: '/app/siem#/hosts/allHosts',
+  hash: '',
+  search: '',
+  state: '',
+};
+
 describe('HeaderGlobal', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
```

Comment:
I'd add a afterAll() to do a reset. Appreciate isolating it from others before you, but you probably should ensure you're cleaned up after your tests are done.

Reply:
It's more accurate to say "we don't need `jest.resetAllMocks()` here because there is only one test." Were there two similar tests, we would be aggregating state on the spy object and there would be the potential for that shared state to affect these tests. However, as mentioned above, a `jest.mock` will only affect the test file in which it's used.

We __really__ should have a top-level `beforeEach(() => jest.resetAllMocks())` to ensure that there's no shared state between any two tests. I would also avoid `beforeAll/afterAll` due to their ability to introduce shared state; we should only use those to explicitly introduce shared state in our tests. But unless the task is prohibitively expensive, we should prefer the `Each` equivalent.

I'm happy to explore this if others agree it's a worthwhile avenue. I would _love_ to remove some of the aforementioned antipatterns and replace all the mocking boilerplate with some higher-level configuration.

Reply:
Thanks for the info. I will keep using `beforeEach` and avoid the `beforeAll`, `afterAll`. Less boiler plate code is always welcome.

Reply:
Thanks @rylnd! I've removed the `spy` and cleaned up the mock as Frank mentioned below. I knew mocks were scoped per file, but was not aware of `spy`'s leaking, so thanks for the knowledge! I'm on board for whatever we can do to clean up these anti-patterns and lower the friction of test-writing! 🙂